node-repository/src/main/java/com/yahoo/vespa/hosted/provision/maintenance/Rebalancer.java

```diff
@@ -35,7 +35,7 @@ public Rebalancer(Deployer deployer,
     @Override
     protected boolean maintain() {
         boolean success = true;
-        if ( ! nodeRepository().zone().getCloud().allowHostSharing()) return success; // Rebalancing not necessary
+        if (nodeRepository().zone().getCloud().dynamicProvisioning()) return success; // Rebalancing not necessary
```

Comment:
"not necessary" is not correct - we hope to enable rebalancing even with dynamic provisioning, but it's a todo.